pkg/ui/userinterface.go

```diff
@@ -0,0 +1,55 @@
+package ui
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/snyk/go-application-framework/pkg/utils"
+)
+
+type UserInterface interface {
+	Output(output string) error
+	OutputError(err error) error
+	NewProgressBar() ProgressBar
+	Input(prompt string) (string, error)
+}
+
+func DefaultUi() UserInterface {
+	// Default Console UI should not have errors (this is tested in consoleui_test.go)
+	return utils.ValueOf(NewConsoleUiBuilder().Build())
+}
+
+type consoleUi struct {
+	writer      io.Writer
+	errorWriter io.Writer
+	reader      *bufio.Reader
+}
+
+func (ui *consoleUi) Output(output string) error {
+	return utils.ErrorOf(fmt.Fprintln(ui.writer, output))
+}
+
+func (ui *consoleUi) OutputError(err error) error {
+	return utils.ErrorOf(fmt.Fprintln(ui.errorWriter, "Error: "+err.Error()))
+}
+
+func (ui *consoleUi) NewProgressBar() ProgressBar {
+	return newProgressBar(ui.writer)
```

Comment:
Question: Are you sure that the writer is where the progress should go? For a console progress, I think it might need to go to stderror, since stdout contains the payload data. Maybe double check with the legacy CLI.